soroban-token-spec/src/lib.rs

```diff
@@ -106,3 +107,35 @@ impl Token {
         panic!("calling into interface");
     }
 }
+
+/// Obtains the full spec for the Token contract
+pub fn get_token_contract_spec_xdr() -> [u8; 2232] {
+    // TODO: We should probably update the contractimpl macro to automatically generate this as a method
+    concat_arrays!(
```

Comment:
I have just given this a try and it's a nightmare (leaving maintainability aside).

It would imply writing 23 while loops and keeping track of the final array size manually (in order to declare it)

 Maybe you had a better way to do this in mind, but, otherwise, it's probably better to expand `contractimpl` (or create another macro).

Reply:
I agree, in general const functions feel like going back in time. I'm taking a look. 👀

Reply:
(Having the `contractimpl` is problematic because macros are stateless and cannot coordinate, so it adds yet another area of code that prevents us from fixing this issue #268.)

Reply:
I think I found a better way, I will push the change soon.

Reply:
I pushed a couple commits to the branch that avoid the multiple while loops. It isn't pretty, but I think it works. I didn't finish filling out the test, I can circle back on that I've just gotta go AFK for a little while.